addons/spreadsheet/static/src/chart/plugins/odoo_chart_ui_plugin.js

```diff
@@ -139,15 +164,24 @@ export default class OdooChartUIPlugin extends UIPlugin {
      */
     _setupChartDataSource(chartId) {
         const dataSourceId = this._getOdooChartDataSourceId(chartId);
-        const definition = this.getters.getChart(chartId).getDefinitionForDataSource();
         if (!this.dataSources.contains(dataSourceId)) {
-            this.dataSources.add(dataSourceId, ChartDataSource, definition);
+            const definition = this.getters.getChart(chartId).getDefinitionForDataSource();
+            this._updateChartDataSourceDefinition(chartId, definition);
         }
         this._setChartDataSource(chartId);
     }
 
     /**
-     * Sets the catasource on the corresponding chart
+     * Sets the datasource on the corresponding chart
+     * @param {string} chartId
+     */
+    _updateChartDataSourceDefinition(chartId, chartDefinition) {
+        const dataSourceId = this._getOdooChartDataSourceId(chartId);
+        this.dataSources.add(dataSourceId, ChartDataSource, chartDefinition);
```

Comment:
That would be more "reset" than update.
I would get the definition back from the core plugin here rather than having it from a parameter, just to be sure they are the same.
```suggestion
    _resetDataSource(chartId) {
        const definition = this.getters.getChart(chartId).getDefinitionForDataSource();
        const dataSourceId = this._getOdooChartDataSourceId(chartId);
        this.dataSources.add(dataSourceId, ChartDataSource, definition);
```